### PR TITLE
3506 - 2

### PR DESCRIPTION
**Pull request by als6k**
## Номер
3506

## Номер задания
2

## Ссылка на видео с демо
https://youtu.be/S8VrC0nWVhQ

### CODE REVIEW

File: 3506/2/Gemfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+source 'https://rubygems.org'
+
+gem 'haversine'
```
Comment (andreydro) [inline 3506/2/Gemfile]: Please lock versions of your gems. It will save you from app break in future

File: 3506/2/checkin.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    if error
+      respond_with :message, text: error
+    else
+      upload_checkin_photo
+    end
+  end
+
+  def validate_checkin
+    return 'Ты не зарегистрировался.' unless user
+    return 'Ты уже в лагере.' if user.in_camp == 'true'
```
Comment (andreydro) [inline 3506/2/checkin.rb]: If you set boolean at first place as value of `in_camp` key, you will not need that weird check. You `user.in_camp` will return `true` boolean

File: 3506/2/checkout.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    if error
+      respond_with :message, text: error
+    else
+      upload_checkout_photo
+    end
+  end
+
+  def validate_checkout
+    return 'Ты не зарегистрировался.' unless user
+    return 'Ты не в лагере.' if user.in_camp == 'false'
```
Comment (andreydro) [inline 3506/2/checkout.rb]: The same with boolean value

File: 3506/2/telegram_bot.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+require_relative 'webhooks_controller'
+
+logger = Logger.new(STDOUT)
+
+BOT_TOKEN = ENV['BOT_TOKEN']
+STUDENT_NUMBERS = YAML.load_file('numbers.yml')['numbers']
+
+Ohm.redis = Redic.new('redis://127.0.0.1:6379')
+
+# export BOT_TOKEN='868196828:AAFzRou1zfvuePNiqIQz-en4F0Ez9t_5Suk'
```
Comment (andreydro) [inline 3506/2/telegram_bot.rb]: Do we need it here?

File: 3506/2/Gemfile (hunk 1 above)
Comment (als6k) [reply 3506/2/Gemfile]: Gems are locked in Gemfile.lock and it's commited into repository. Even if gem updated in the future, bundle install will install the version specified in Gemfile.lock

**Pull request merged**